homeassistant/config_entries.py

```diff
@@ -1015,17 +1020,57 @@ async def async_reload(self, entry_id: str) -> bool:
         """Reload an entry.
 
         If an entry was not loaded, will just load.
+
+        Returns True if the entry was reloaded, or
+        a reload was successfully scheduled.
         """
         if (entry := self.async_get_entry(entry_id)) is None:
             raise UnknownEntry
 
-        async with entry.reload_lock:
-            unload_result = await self.async_unload(entry_id)
+        ratelimit = self._reload_ratelimit
+        now = dt_util.utcnow()
+
+        @callback
+        def _async_future_reload() -> None:
+            assert entry is not None
+            asyncio.create_task(self._async_future_reload(entry))
```

Comment:
Probably should not hold a strong reference to the entry or only hold the entry id in case it gets unloaded while waiting for cooldown